tests/test_greedy_sampler.py

```diff
@@ -280,3 +280,36 @@ def test_initial_states_sample_like(self, initial_states):
         # result is identical to initial state, with zero downhill moves
         np.testing.assert_array_equal(ss.record.sample, [[-1, -1]])
         np.testing.assert_array_equal(ss.record.num_steps, [0])
+
+
+class TestTimingInfo(unittest.TestCase):
+    def setUp(self) -> None:
+        empty = dimod.BQM(dimod.SPIN)
+        one = dimod.BQM.from_ising({"a": 1}, {})
+        two = dimod.BQM.from_ising({}, {("abc", (1, 2)): -1})
+
+        sampler = SteepestDescentSampler()
+        rng = np.random.default_rng(59921)
+
+        self.sample_sets = []
+        for bqm in [empty, one, two]:
+            sample_set = sampler.sample(bqm, seed=rng.integers(2**30))
+            self.sample_sets.append(sample_set)
+
+        self.timing_keys = ["preprocessing_ns", "postprocessing_ns", "sampling_ns"]
+
+    def test_keys_exist(self):
+        for sample_set in self.sample_sets:
+            self.assertIn("timing", sample_set.info)
+            timing = sample_set.info['timing']
+            for timing_key in self.timing_keys:
+                self.assertIn(timing_key, timing)
```

Comment:
You could use `issubset` here, and mark subtests:
```suggestion
        self.timing_keys = set(["preprocessing_ns", "postprocessing_ns", "sampling_ns"])

    def test_keys_exist(self):
        for sample_set in self.sample_sets:
            with self.subTest(ss=sample_set):
                self.timing_keys.issubset(sample_set.info['timing'])
```

Reply:
Thanks, that's much more concise! Can I make the same changes for simulated annealing's test in *this* PR?

Reply:
Sure, why not.